### PR TITLE
Add div and mult functions

### PR DESCRIPTION
**Pull request by svozza**
Last of the functions from #127. I never want to see another `concat` error message again!

### CODE REVIEW

**Comment by davidchambers** [comment]
> I never want to see another `concat` error message again!

:stuck_out_tongue_winking_eye:

File: index.js [hunk 1]
```diff
@@ -2277,6 +2278,35 @@
       [$.FiniteNumber, $.FiniteNumber],
       function(a) { return a - 1; });
 
+  //# mult :: FiniteNumber -> FiniteNumber -> FiniteNumber
+  //.
+  //. Returns the product of two (finite) numbers.
+  //.
+  //. ```javascript
+  //. > S.mult(2, 2)
```
Comment (davidchambers) [inline index.js]: Let's use values with don't also *add* to give 4. ;)

File: index.js [hunk 2]
```diff
@@ -2277,6 +2278,35 @@
...
+  //. 4
+  //. ```
+  S.mult =
+  def('mult',
+      {},
+      [$.FiniteNumber, $.FiniteNumber, $.FiniteNumber],
+      function(a, b) { return a * b; });
+
+  //# div :: FiniteNumber -> FiniteNumber -> FiniteNumber
```
Comment (davidchambers) [inline index.js]: The second argument's type is `NonZeroFiniteNumber`.

File: test/index.js [hunk 3]
```diff
@@ -3374,6 +3382,100 @@ describe('number', function() {
...
+                            '‘mult’ expected a value of type FiniteNumber ' +
+                            'as its second argument; received Infinity'));
+
+      assert.throws(function() { S.mult(2, -Infinity); },
+                    errorEq(TypeError,
+                            '‘mult’ expected a value of type FiniteNumber ' +
+                            'as its second argument; received -Infinity'));
+    });
+
+    it('mults two numbers', function() {
```
Comment (davidchambers) [inline test/index.js]: `s/mults/multiplies/`

File: index.js (hunk 1 above)
Comment (svozza) [inline index.js]: Ha. Good point.

File: index.js (hunk 2 above)
Comment (svozza) [inline index.js]: :zap:

File: test/index.js (hunk 3 above)
Comment (svozza) [inline test/index.js]: :zap:

File: index.js [hunk 4]
```diff
@@ -2277,6 +2278,35 @@
...
+      [$.FiniteNumber, $.FiniteNumber, $.FiniteNumber],
+      function(a, b) { return a * b; });
+
+  //# div :: FiniteNumber -> NonZeroFiniteNumber -> FiniteNumber
+  //.
+  //. Returns the result of dividing its first argument (a finite number) by
+  //. its second argument (a non-zero finite number).
+  //.
+  //. ```javascript
+  //. > S.div(4, 2)
```
Comment (davidchambers) [inline index.js]: The same logic applies here, actually. Let's use arguments which result in different values for `R.div` and `R.sub`. Also, let's make it clear that this is not integer division.

Comment (svozza) [inline index.js]: Fair enough, although regarding making it clear that this isn't integer division, I'm not really sure how much clearer we can make it given the type signature at the top of the function description.

Comment (davidchambers) [inline index.js]: I was thinking something like `S.div(7, 2) = 3.5` (rather than `3` as would be the case with integer division).

Comment (svozza) [inline index.js]: :zap:

File: index.js [hunk 5]
```diff
@@ -2277,6 +2278,38 @@
...
+  //. Returns the result of dividing its first argument (a finite number) by
+  //. its second argument (a non-zero finite number).
+  //.
+  //. ```javascript
+  //. > S.div(8, 2)
+  //. 4
+  //.
+  //. > S.div(7, 2)
+  //. 3.5
+  //. ```
```
Comment (davidchambers) [inline index.js]: The information content of the second example is a strict superset of the information content of the first example, so we could remove the first example.

Comment (svozza) [inline index.js]: :zap:

File: index.js [hunk 6]
```diff
@@ -2277,6 +2278,36 @@
...
+      [$.FiniteNumber, $.FiniteNumber, $.FiniteNumber],
+      function(a, b) { return a * b; });
+
+  //# div :: FiniteNumber -> NonZeroFiniteNumber -> FiniteNumber
+  //.
+  //. Returns the result of dividing its first argument (a finite number) by
+  //. its second argument (a non-zero finite number).
+  //.
+  //. ```javascript
+  //.
```
Comment (davidchambers) [inline index.js]: :space_invader:

Comment (svozza) [inline index.js]: :rocket:

**Comment by davidchambers** [comment]
:deciduous_tree:

**Pull request merged**